backend/server/src/role.rs

```diff
@@ -105,3 +108,42 @@ pub async fn new_role(
     })
     .await
 }
+
+#[derive(Serialize)]
+pub enum QuestionsError {
+    RoleNotFound,
+    CampaignNotFound,
+    Unauthorized,
+}
+
+#[get("/<role_id>/questions")]
+pub async fn get_questions(
+    role_id: i32,
+    user: User,
+    db: Database,
+) -> Result<Json<GetQuestionsResponse>, Json<QuestionsError>> {
+    // First check that the role is valid and the user should be able to access the ids.
+    // We can't use the helper function below since behaviour depends on the draft
+    // status of the campaign.
+
+    db.run(move |conn| {
+        let role = Role::get_from_id(conn, role_id).ok_or(Json(QuestionsError::RoleNotFound))?;
+        let campaign = Campaign::get_from_id(conn, role.campaign_id)
+            .ok_or(Json(QuestionsError::CampaignNotFound))?;
+
+        // Prevent people from viewing while it's in draft mode,
+        // unless they have adequate permissions
+        OrganisationUser::campaign_admin_level(campaign.id, user.id, &conn)
```

Comment:
role_admin_level() exists in organisationUser - could just use that

Reply:
True, but we need campaign anyway to use campaign.published in the check.

Reply:
fair enough, lgtm otherwise :)